app/server.js

```diff
@@ -16,6 +17,32 @@ app.use(express.static("public"))
 
 
 // TODO: everything
+const tokenStorage = {};
+
+const authentication = (req, res, next) => {
+    const authHeader = req.headers['authorization'];
+    let token = authHeader && authHeader.split(' ')[1];
+
+    if (!token || !tokenStorage[token]) {
+        return res.status(401).json({ error: "Token required."});
+    }
+    next();
+};
+
+const generateRandomToken = () => {
+    return crypto.randomBytes(32).toString("hex");
+}
+
+app.get("/api/auth/demoKey", (req, res) => {
+    const token = generateRandomToken();
+    tokenStorage[token] = true;
+    console.log(tokenStorage); //keep track of tokens created
+    return res.json({ token });
+})
+
+app.get("/pos/", authentication, (req, res) => {
```

Comment:
so since another pr is using the /pos route now, we should change this to `"/api/auth/test"` or something